### PR TITLE
DAOS-16482 control: Increase min hugepages for single tgt count

### PR DESCRIPTION
**Pull request by tanabarr**
DMA buffer allocations may fail for single target count due to
insufficient hugepages available. Increase minimum number of hugepages
for single target count when in PMem mode.

Feature: control
Required-githooks: true

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by knard-intel — DISMISSED** [review]

File: src/control/server/storage/bdev.go [hunk 1]
```diff
@@ -38,7 +38,8 @@ const (
 	NilBdevAddress = "<nil>"
 	sysXSTgtID     = 1024
 	// Minimum amount of hugepage memory (in bytes) needed for each target.
-	memHugepageMinPerTarget = 1 << 30 // 1GiB
+	memHugepageMinPerTarget      = 1 << 30 // 1GiB
+	minNrTargetsForCalcHugepages = 2
```
Comment (kjacque) [inline src/control/server/storage/bdev.go:42]: It would be nice to have a comment here explaining the reason this is needed.

**Review by kjacque — DISMISSED** [review]
Logic seems fine based on the discussion in standup. I wouldn't block on my comment below, but I do think some documentation is merited in the code in this case.

**Review by knard-intel — APPROVED** [review]

**Comment by tanabarr** [comment]
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-15115/5/tests shows that the single target test variants now pass when they were failing in run no. 4.

**Comment by tanabarr** [comment]
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-15115/5/testReport/ failed in ConfigGenerate tests related to https://daosio.atlassian.net/browse/DAOS-16334 known issue

**Review by mjmac — APPROVED** [review]

**Pull request merged**